internal/stanza/emitter.go

```diff
@@ -17,46 +17,168 @@ package stanza // import "github.com/open-telemetry/opentelemetry-collector-cont
+	return LogEmitterOption(func(le *LogEmitter) {
+		le.flushInterval = flushInterval
+	})
+}
+
+// LogEmitterWithLogger returns an option that makes the LogEmitter use the specified logger
+func LogEmitterWithLogger(logger *zap.SugaredLogger) LogEmitterOption {
+	return LogEmitterOption(func(le *LogEmitter) {
+		le.OutputOperator.BasicOperator.SugaredLogger = logger
+	})
 }
 
 // NewLogEmitter creates a new receiver output
-func NewLogEmitter(logger *zap.SugaredLogger) *LogEmitter {
-	return &LogEmitter{
+func NewLogEmitter(opts ...LogEmitterOption) *LogEmitter {
+	le := &LogEmitter{
 		OutputOperator: helper.OutputOperator{
 			BasicOperator: helper.BasicOperator{
 				OperatorID:    "log_emitter",
 				OperatorType:  "log_emitter",
-				SugaredLogger: logger,
+				SugaredLogger: zap.NewNop().Sugar(),
 			},
 		},
-		logChan: make(chan *entry.Entry),
+		logChan:       make(chan []*entry.Entry),
+		maxBatchSize:  defaultMaxBatchSize,
+		batch:         make([]*entry.Entry, 0, defaultMaxBatchSize),
+		flushInterval: defaultFlushInterval,
+		cancel:        func() {},
 	}
-}
 
-// Process will emit an entry to the output channel
-func (e *LogEmitter) Process(ctx context.Context, ent *entry.Entry) error {
-	select {
-	case e.logChan <- ent:
-	case <-ctx.Done():
+	for _, opt := range opts {
+		opt(le)
 	}
+
+	return le
+}
+
+// Start starts the goroutine(s) required for this operator
+func (e *LogEmitter) Start(_ operator.Persister) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	e.cancel = cancel
+
+	e.wg.Add(1)
+	go e.flusher(ctx)
 	return nil
 }
 
-// Stop will close the log channel
+// Stop will close the log channel and stop running goroutines
 func (e *LogEmitter) Stop() error {
 	e.stopOnce.Do(func() {
+		e.cancel()
+		e.wg.Wait()
+
 		close(e.logChan)
 	})
+
 	return nil
 }
+
+// Process will emit an entry to the output channel
+func (e *LogEmitter) Process(ctx context.Context, ent *entry.Entry) error {
+	oldBatch := e.appendEntry(ent)
+
+	if len(oldBatch) > 0 {
+		e.flush(ctx, oldBatch)
+	}
+
+	return nil
+}
+
+// appendEntry appends the entry to the current batch. If maxBatchSize is reached, a new batch will be made, and the old batch
+// (which should be flushed) will be returned
+func (e *LogEmitter) appendEntry(ent *entry.Entry) []*entry.Entry {
+	e.batchMux.Lock()
+	defer e.batchMux.Unlock()
+
+	e.batch = append(e.batch, ent)
+	if uint(len(e.batch)) >= e.maxBatchSize {
+		var oldBatch []*entry.Entry
+		oldBatch, e.batch = e.batch, make([]*entry.Entry, 0, e.maxBatchSize)
+		return oldBatch
+	}
+
+	return nil
+}
+
+// flusher flushes the current batch every flush interval. Intended to be run as a goroutine
+func (e *LogEmitter) flusher(ctx context.Context) {
+	defer e.wg.Done()
+
+	ticker := time.NewTicker(e.flushInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			oldBatch := e.makeNewBatch()
+
+			if len(oldBatch) > 0 {
+				e.flush(ctx, oldBatch)
+			}
```

Comment:
I think this regressed when you reverted the latest change.

```suggestion
			if oldBatch := e.makeNewBatch(); len(oldBatch) > 0 {
				e.flush(ctx, oldBatch)
			}
```